string text
uint32 ctr
